lib/util.js

```diff
@@ -501,16 +501,22 @@ function formatObject(ctx, value, recurseTimes, visibleKeys, keys) {
 
 function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
-  for (var i = 0, l = value.length; i < l; ++i) {
-    if (hasOwnProperty(value, String(i))) {
-      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-          String(i), true));
-    } else {
-      output.push('');
+  var isSparse = keys.length * 10 < value.length;
```

Comment:
@thefourtheye added a comment. If less than 1/10th of the slots in an array are populated, `util.inspect` / `console.log` now renders it compactly

```
var arr = []
arr[100] = 1
console.log(arr)
```

**Before:**
`[ , , , , , (lots of commas) , 1 ]`

**After:**
`[ 100:1 ]`